### PR TITLE
docs(tree, tree-item): fix slot documentation

### PR DESCRIPTION
**Pull request by benelan**
**Related Issue:** #2807

## Summary
Fixed the tree and tree-item slot documentation pointed out by @jcfranco during verification.

### CODE REVIEW

File: src/components/calcite-tree-item/calcite-tree-item.tsx [hunk 1]
```diff
@@ -18,6 +18,10 @@ import { Scale } from "../interfaces";
 import { CSS, SLOTS, ICONS } from "./resources";
 import { CSS_UTILITY } from "../../utils/resources";
 
+/**
+ * @slot children - A slot for adding nested `calcite-tree` elements.
+ * @slot - A slot for adding content to the item.
```
Comment (jcfranco) [inline src/components/calcite-tree-item/calcite-tree-item.tsx:23]: Sidebar: should we list the default slot first (general to specific)?

**Review by jcfranco — APPROVED** [review]
👍🏼

**Comment by jcfranco** [comment]
@caripizza Can you give this a quick look? 👀

**Review by caripizza — APPROVED** [review]
Thanks for fixing up these docs @benelan LGTM! 👍

File: src/components/calcite-tree-item/calcite-tree-item.tsx (hunk 1 above)
Comment (benelan) [reply src/components/calcite-tree-item/calcite-tree-item.tsx:23]: Sure I can do that as part of #2974

**Pull request merged**